packages/babel-traverse/src/scope/index.js

```diff
@@ -952,11 +951,11 @@ export default class Scope {
    */
 
   getAllBindings(): Object {
-    const ids = Object.create(null);
+    let ids = Object.create(null);
 
     let scope = this;
     do {
-      defaults(ids, scope.bindings);
+      ids = { ...scope.bindings, ...ids };
```

Comment:
```suggestion
      for (const key of Object.keys(scope.bindings)) {
        if (key in ids === false) {
          ids[key] = scope.bindings[key];
        }
      }
```

This is probably a the safer replacement as it ensures that ids (the return value) keeps the null prototype.  From what I can tell `scope.bindings` should was also created by `Object.create(null)` so this will ensure prototype pollution cannot alter the result.